### PR TITLE
Mlflow registry

### PR DESCRIPTION
**Pull request by mobiuscreek**
Adds a very basic class that provides an interface to the mlflow registry:

* The `list_experiments` and `list_models` functions both return a list of logged experiments and registered models respectively.
* The function  `register_model()` registers the most recent and better model (if that's the case). This functionality can be overriden.
* The `load_model` function may not be needed here and  it only works for fastai models.
* Additional tests

Things that don't work:
* No exception handling
* In a lot of cases the query argument for filtering has to be carefully thought out.

### CODE REVIEW

**Review by metazool — CHANGES_REQUESTED** [review]
I pushed up a failing test for `register_model` intended to show the `IndexError` in the case there's no previous run to compare to. Probably overkill to probe much further into what is essentially an edge case / worry too much about the metrics keys until we've used this a bit more, but improves the test coverage? Let me know what you think of it 

The mlflow query by tag was picky about how the tag is sent, wanted the tag name quoted within the string, so I made a slight change there.

Otherwise this looks good, useful, solid work, it will still run as-is and I'll try putting it through the `label-wash`

File: corescore/mlflowregistry.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    def register_model(
+            self,
+            query,
+            metric,
+            override=False,
+            name="latest-reg-model"):
+        """Register a model only if it shows better performance"""
+        search_result = self.list_experiments(query=query)
+        old_metric = search_result[1]['data']['metrics'][f"{metric}"]
```
[the remark this answers is not in the document]
Comment (mobiuscreek) [reply corescore/mlflowregistry.py]: I added the try except block but I'll change it to a warning and register the model anyway like you said. Thanks for looking at it!

**Comment by mobiuscreek** [comment]
I changed the `register_model` function. It now always adds the latest model to the registry and throws an exception if the query doesn't find any model (returns an empty list). I added a mock response that replicates it and added a test that checks that the exception is raised correctly.

**Comment by metazool** [comment]
I'd forgotten this was still outstanding; closing it because we have the improvements in #10
